### DISCUSSION TITLE
Debug graphics API

### DISCUSSION THREAD

**Discussion by local-interloper** [godotengine/godot-proposals#8776] [3D]
Coming from Unity and Unreal, visual debugging was always handy. It would be nice if we could have an API for drawing simple wireframe shapes such as boxes, capsules, spheres and lines in the viewport. There are some plugins that allow that out there already but this should really be a engine feature.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/5196.
